File: src/services/MetricsService/Metric.ts
export class Metric {
  readonly timestamp: number;
  readonly identifier: string;
  readonly region: string;
  readonly payload: [string, string][];

  constructor(timestamp: number, identifier: string, region: string, payload: [string, string][]) {
    this.timestamp = timestamp;
    this.identifier = identifier;
    this.region = region;
    this.payload = payload;
  }
}

File: ios/Metric.m
//
//  Metric.m
//  CovidShield
//
//  Created by Clement Janin on 2021-04-28.
//

#import "Metric.h"

@interface Metric ()

@property (strong, nonatomic) NSNumber *timestamp;
@property (strong, nonatomic) NSString *identifier;
@property (strong, nonatomic) NSString *region;

@end

@implementation Metric

- (instancetype)initWithTimestamp:(NSNumber *)timestamp identifier:(NSString *)identifier region:(NSString *)region
{
  self = [super init];
  if (self) {
    self.timestamp = timestamp;
    self.identifier = identifier;
    self.region = region;
  }
  return self;
}

@end
